docs/app-host/persistent-containers.md

```diff
@@ -0,0 +1,106 @@
+
+builder.AddProject<Projects.InventoryService>("inventory")
+       .WithReference(db);
+
+builder.Build().Run();
+```
+
+In the preceding example, the PostgreSQL container is configured to persist between app host runs, while the `inventory` project references the database as normal.
+
+## Dashboard visualization
+
+The .NET Aspire dashboard shows persistent containers with a distinctive pin icon (📌) to help you identify them:
+
+:::image type="content" source="../whats-new/media/persistent-container.png" lightbox="../whats-new/media/persistent-container.png" alt-text="Screenshot of the .NET Aspire dashboard showing a persistent container with a pin icon.":::
+
+After the app host stops, persistent containers continue running and can be seen in your container runtime (such as Docker Desktop):
+
+:::image type="content" source="../whats-new/media/persistent-container-docker-desktop.png" lightbox="../whats-new/media/persistent-container-docker-desktop.png" alt-text="Screenshot of Docker Desktop showing a persistent RabbitMQ container still running after the app host stopped.":::
+
+## Configuration change detection
+
+Persistent containers are automatically recreated when the app host detects meaningful configuration changes. .NET Aspire tracks a hash of the configuration used to create each container and compares it to the current configuration on subsequent runs. If the configuration differs, the container is recreated with the new settings.
+
+This mechanism ensures that persistent containers stay synchronized with your app host configuration without requiring manual intervention.
+
+## Container naming and uniqueness
+
+By default, persistent containers use a naming pattern that combines:
+
+- The service name you specify in your app host.
+- A postfix based on a hash of the app host project path.
+
+This naming scheme ensures that persistent containers are unique to each app host project, preventing conflicts when multiple .NET Aspire projects use the same service names.
+
+For example, if you have a service named `"postgres"` in an app host project located at `/path/to/MyApp.AppHost`, the container name might be `postgres-abc123def` where `abc123def` is derived from the project path hash.
+
+### Custom container names
+
+For advanced scenarios, you can set a custom container name using the <xref:Aspire.Hosting.ContainerResourceBuilderExtensions.WithContainerName*> method:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var postgres = builder.AddPostgres("postgres")
+                      .WithLifetime(ContainerLifetime.Persistent)
+                      .WithContainerName("my-shared-postgres");
+
+builder.Build().Run();
+```
+
+When you specify a custom container name, .NET Aspire can reuse an existing container that wasn't created by the app host. Such containers are never automatically recreated, even if the app host configuration changes, giving you full control over their lifecycle.
```

Comment:
We should probably note that Aspire will still create a new persistent container if there isn't an existing one with the given custom name.